libs/components/indicators/testing/src/icon/icon-harness.ts

```diff
@@ -26,15 +26,15 @@ export class SkyIconHarness extends SkyComponentHarness {
 
   /** Gets the icon name */
   public async getIconName(): Promise<string | undefined> {
-    const iconClass = await (await this.#getIcon()).getAttribute('class');
-    const regex = /^sky-i-|fa-(?!fw|2xs|xs|sm|lg|2xl|[0-9]+x)/;
-    let className = iconClass
-      ?.split(' ')
-      .find((classValue: string) => classValue.match(regex));
-
-    className = className?.replace(/-line|-solid$/, '');
-    className = className?.replace(/^sky-i-|fa-/, '');
-    return className;
+    const iconClass = await (await this.#getIcon()).getProperty('classList');
+    for (let i = 0; i < iconClass.length; i++) {
```

Comment:
One last thing to try: use `for...of` instead of a `for` loop here. It'll keep you from having to access the icon class by index multiple times inside the `for` loop.

```
const iconClasses = await (await this.#getIcon()).getProperty('classList');
for (const iconClass of iconClasses) {
  ...
}
```

Reply:
well damn, when i had iconClass hard cased to DOMTokenLlist it didnt like me using `for...of` loops bc it said DOMTokenList doesn't have an iterator, it works now tho without the type being set for `iconClass`